### PR TITLE
Line class refactoring & unit tests

### PR DESCRIPTION
**Pull request by rzats**
### Contains

- Extracts the coordinate computations from `Canvas#drawLine` into a separate helper method in the existing `Line` class and adds some unit tests for the new method.

### How to test

- Use the BT editor to check that line rendering is working properly.
- Verify that the new tests are not failing and that their expected/actual values are appropriate.

### Outstanding before merging

No outstanding issues.

### CODE REVIEW

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

File: engine-tests/src/test/java/org/terasology/rendering/nui/internal/LineTest.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+            Line.getLineCoordinates(20, 5, 20, 35, Rect2i.EMPTY, cropRegion));
+        assertEquals(new Line.LineCoordinates(new Vector2i(20, 25), new Vector2i(30, 30)),
+            Line.getLineCoordinates(20, 25, 40, 40, Rect2i.EMPTY, cropRegion));
+    }
+
+    /**
+     * @return A random integer ranging from 1 to 5 (inclusive)
+     */
+    private int shortRnd() {
+        return (int) (Math.random() * 5) + 1;
```
Comment (flo) [inline engine-tests/src/test/java/org/terasology/rendering/nui/internal/LineTest.java]: I would strongly recommend not to use random in unit tests. You will never know that way if it truly will always pass or if you went just lucky in the test runs you made.

**Comment by flo** [comment]
Very cool, that you also made some tests

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by flo** [comment]
thanks for adding some tests :)

**Pull request merged**